From: Russell King <rmk+kernel@armlinux.org.uk>
Date: Tue, 12 Jul 2016 00:04:13 +0100
Subject: [PATCH] net: mvneta: enable flow control for fixed connections

Allow symetric flow control to be enabled for fixed link connections as
well as other types of connections by setting the supported and
advertised capability bits.

Signed-off-by: Russell King <rmk+kernel@armlinux.org.uk>
---

--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -3144,6 +3144,8 @@ static void mvneta_validate_support(stru
 	phylink_set(mask, BNC);
 	phylink_set(mask, Backplane);
 
+	/* Asymmetric pause is unsupported */
+	phylink_set(mask, Pause);
 	/* Half-duplex at speeds higher than 100Mbit is unsupported */
 	phylink_set(mask, 1000baseT_Full);
 	phylink_set(mask, 1000baseX_Full);
@@ -3156,9 +3158,6 @@ static void mvneta_validate_support(stru
 		phylink_set(mask, 100baseT_Full);
 	}
 
-	if (mode != MLO_AN_FIXED)
-		phylink_set(mask, Pause);
-
 	bitmap_and(support, support, mask, __ETHTOOL_LINK_MODE_MASK_NBITS);
 }
 
